### PR TITLE
Allow using empty strings as job parameters

### PR DESCRIPTION
**Pull request by kanterov**
## Changes
Allow using an empty string as the "default" value for job parameters, otherwise, such jobs are rejected because "default" field is required.

Remove "omitempty" modifier because both fields are required.

## Tests

Verified manually with Terraform

- [x] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Review by gaborratky-db — APPROVED** [review]

**Comment by pietern** [comment]
@kanterov @gaborratky-db Isn't it valid to define a job where some job parameters are required? I.e. you have to specify the parameters when running it? If so, then this approach does not allow that use case.

I believe this fix needs to use the `ForceSendFields` feature to mark the default as set or not. That way it is possible to both define parameters with no default (such that specifying them for a run is mandatory) as well as parameters with the default being an empty string.

**Comment by kanterov** [comment]
@pietern, we don't really have 'required' job parameters in this case. But, the 'default' field in the API is required. If it's set to an empty string, it gets skipped in the API request because of the 'omitempty' thing, and that's causing a 400 error. So, we've got to make sure we include this 'default' field every time, even if it's just an empty string.

**Review by pietern — COMMENTED** [review]
@kanterov Please add a unit test to confirm that empty strings are now specified. There are two cases to consider; a user explicitly passes an empty string (good), and a user doesn't specify the default at all, which would default to an empty string. In the latter case, we could see configuration drift on the TF side.

**Review by lennartkats-db — APPROVED** [review]

**Review by mgyucht — APPROVED** [review]

File: internal/acceptance/job_test.go [hunk 1]
```diff
@@ -90,8 +90,22 @@ func TestAccJobTasks(t *testing.T) {
 
 				notebook_task {
 					notebook_path = databricks_notebook.this.path
+					base_parameters = {
+						"param_0" = "{{job.parameters.empty_default}}"
+						"param_0" = "{{job.parameters.non_empty_default}}"
```
[the remark this answers is not in the document]
Comment (kanterov) [reply internal/acceptance/job_test.go:95]: Oh... my bad, we need to fix it. It shouldn't matter because we don't run it, and I've tried empty value manually. But let's fix the test.
